### ISSUE TITLE
How to train the initial state?

### ISSUE THREAD

**Issue by peterroelants** [Element-Research/rnn#176]
Is it possible to train the initial state of the Recurrent and LSTM modules? I think the initial state for the Recurrent layer can be trained via the start Module given during construction, but what about the LSTM?

**Comment by nicholas-leonard** [comment]
For LSTM it will use the bias of the internal modules.

**Comment by vyouman** [comment]
@nicholas-leonard 
How can I feed the initial state to the LSTM? For example if I have got some image features by pretrained CNN and want to feed it to LSTM to be the initial state. I look through the LSTM code and think maybe I should
https://github.com/Element-Research/rnn/blob/master/LSTM.lua#L142-144
modify the userPrevOutput and userPrevCell. If I've got a lstm instance, then use something like

> lstm.userPrevOutput = torch.Tensor(batchSizeoutputSize):fill(1)
> lstm.userPrevCell = torch.Tensor(batchSizeoutputSize):fill(0.5)

Right? Thanks.
